### PR TITLE
Over-indent multiline interpolations

### PR DESCRIPTION
**Pull request by dralletje**
I made this PR after needing the same features as #12 

It looks at the value in interpolations and adds the indentation to every but the first line (because that one already has the indentation).

I'm not sure if you'd want this in the package, as it assumes a lot more stuff, but on the other hand I can not imagine a scenario where you don't want this 🙃

### CODE REVIEW

File: __tests__/dedent-tests.js [hunk 1]
```diff
@@ -27,6 +27,29 @@ describe("dedent", () => {
...
+      dd`
+        first
+          ${dd`
+            second
+            third
+          `}
+        fourth
+      `
+    ).toMatchSnapshot();
+  });
```
Comment (rattrayalex-stripe) [inline __tests__/dedent-tests.js]: Probably worth adding a test like the one referenced in #12 , namely:

```jsx
const weird = dedent`
  foo.
  ${things.map(x => dedent`
    hello
      world
  `).join('\n')}
  bar.
`;
```
which should produce (as it does): 
```
foo.
hello
  world
hello
  world
hello
  world
bar.
```
Possibly also 
```jsx
const weirder = dedent`
  foo.
    ${things.map(x => dedent`
      hello
        world
    `).join('\n')}
  bar.
`;
```
which I think should produce
```
foo.
  hello
    world
  hello
    world
  hello
    world
bar.
```

Comment (dralletje) [reply __tests__/dedent-tests.js]: I feel these extra tests are not necessary, as this is covered by the 'non-map' version of the tests... dunno how the author feels about tests testing the same thing :P

**Comment by dralletje** [comment]
Added another commit to make a bit more strict

File: __tests__/dedent-tests.js (hunk 1 above)
Comment (rattrayalex-stripe) [reply __tests__/dedent-tests.js]: ... Yes, you're correct; I wasn't reading very carefully. Thanks!

The inner template should probably include indents/dedents, though, eg; 

```jsx
dd`
  first
    ${dd`
         second
       third
           fourth
    `}
  fifth
`
```

to ensure that "inner template" logic is the same as "outer template" logic

Comment (rattrayalex-stripe) [reply __tests__/dedent-tests.js]: (to be clear, I think the above is purely optional and I would expect it to pass today)

**Comment by rattrayalex-stripe** [comment]
cc @dmnd is this mergeable?

**Comment by ZhouHansen** [comment]
Hi! @rattrayalex-stripe you can use [endent](https://github.com/indentjs/endent) to achieve what you want before this pr merged.

**Comment by JoshuaKGoldberg** [comment]
Note: I posted a question in #12 about whether we want this to always be the case, or an opt-in.

**Comment by JoshuaKGoldberg** [comment]
Closing for now to keep the queue small - let's discuss in #12. Cheers!

**Pull request closed without merging**